### DISCUSSION TITLE
AXParameter fields does not automatically appear in web GUI

### DISCUSSION THREAD

**Discussion by onie5** [AxisCommunications/discussions#237] [ACAP]
Using the AXParameter API and adding parameter fields programmatically using the ax_parameter_add() function.

The fields does not show up on the web GUI when app is started. I need to manually refresh the page for the fields to appear.

No error is popping up in the log. Any advice is appreciated.

Thank you.

**Comment by onie5** [comment]
Adding to above information, the parameter field is populated with the resolution list available in the particular camera. The resolution list is pulled using vdo library and provided as an "enum" type argument in the ax_parameter_add() function. Can the loading of the list causes the issue of not refreshing the page at startup?

**Comment by shadysp** [comment]
Can you please let me know:
-Which Firmware version?
-Do you experience the same behaviour with other version?
-Are you using the call back function?

It could be as you mentioned the loading of the list, the order or  sequence of execution might be causing this behaviour.
Can you test adding the parameter and pass Initial value as NULL which will set the parameter to an empty string, then get the resolution list  and update the parameter value using ax_parameter_set.  And the ax_parameter_register_callback() function will run  whenever a parameter value is updated.

**Comment by vivekatoffice** [comment]
Related links:
[axparameter](https://github.com/AxisCommunications/acap-native-sdk-examples/tree/main/axparameter)
[Parameter API](https://axiscommunications.github.io/acap-documentation/docs/api/native-sdk-api.html#parameter-api)


![image](https://github.com/AxisCommunications/discussions/assets/52032125/40e75b06-ee3c-4279-a32a-599d11f94566)
